torch/distributions/kl.py

```diff
@@ -176,10 +176,10 @@ def kl_divergence(p, q):
 
 @register_kl(Bernoulli, Bernoulli)
 def _kl_bernoulli_bernoulli(p, q):
-    t1 = p.probs * (p.probs / q.probs).log()
+    t1 = p.probs * (torch.nn.functional.softplus(-q.logits) - torch.nn.functional.softplus(-p.logits))
```

Comment:
This is slightly inefficient I think. For numerical stability, you can try rewriting the entire expression as:
`p.probs * p.logits - p.probs * q.logits + (-p.probs).log1p() - (-q.probs).log1p()`.

However, if you are specifying `logits` alone, then the `logits_to_probs` will be called when the `probs` attribute is used.

Reply:
`p.probs * (p.logits - q.logits) + (-p.probs).log1p() - (-q.probs).log1p()` is even better. Besides we call `.probs` many times anyway. But in this way should be more stable.